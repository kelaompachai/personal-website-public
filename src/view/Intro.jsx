import React from 'react';
import text from '../model/intro';
import profilePhoto from '../../assets/profile-photo.jpg';

function Intro() {
  return (
    <div className="flex text-lg font-light">
      <img className="rounded-full mx-7 object-contain size-36" src={profilePhoto} alt="My beautiful face"></img>
      <p className="mx-7">{text}</p>
      <div className="flex flex-col items-center">
        <span>JosephLPaul@proton.me</span>
        <a className="text-well-read-400 underline" href="https://www.github.com/kelaompachai">Github</a>
        <a className="text-well-read-400 underline" href="https://www.linkedin.com/in/joseph-lawrence-paul">LinkedInLink</a>
      </div>
    </div>
  );
}

export default Intro;
