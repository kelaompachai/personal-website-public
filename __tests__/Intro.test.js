import { render, screen } from '@testing-library/react';
import Intro from '../src/view/Intro';

it('looks like my Intro component', () => {
  const container = render(<Intro />);
  expect(container.baseElement).toMatchSnapshot();


  const image = screen.getByRole('img');

  // checking image element
  expect(image).toHaveAttribute('src', 'testFileStub');
  expect(image).toHaveAttribute('alt', 'My beautiful face');

  // checking that intro text is present
  expect(screen.queryByText(/Hi, I'm Joe Paul/)).toBeInTheDocument();

  // checking that my email is present
  expect(screen.queryByText(/jobs.alarm618@passinbox.com/i)).toBeInTheDocument();  

  // checking for github and linkedin
  const github = screen.queryByText(/Github/)
  expect(github).toBeInTheDocument();
  expect(github).toHaveRole('link');
  expect(github).toHaveAttribute('href', 'https://www.github.com/kelaompachai');

  const linkedin = screen.queryByText(/LinkedIn/);
  expect(linkedin).toBeInTheDocument();
  expect(linkedin).toHaveRole('link');
  expect(linkedin).toHaveAttribute('href', 'https://www.linkedin.com/in/joseph-lawrence-paul');
});