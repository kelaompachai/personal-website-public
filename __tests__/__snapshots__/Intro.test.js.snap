// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks like my Intro component 1`] = `
<body>
  <div>
    <div
      class="flex text-lg font-light"
    >
      <img
        alt="My beautiful face"
        class="rounded-full mx-7 object-contain size-36"
        src="testFileStub"
      />
      <p
        class="mx-7"
      >
        Hi, I'm Joe Paul, a software engineer who's passionate about what's true. I enjoy building websites (front and back end), solving leetcode problems, reading classics, and hiking. I'm currently working on building a SQL database with an accompanying UI, learning AWS, and reading Eliot's Daniel Deronda. Feel free to email me if you'd like to talk about any of those things.
      </p>
      <div
        class="flex flex-col items-center"
      >
        <span>
          JosephLPaul@proton.me
        </span>
        <a
          class="text-well-read-400 underline"
          href="https://www.github.com/kelaompachai"
        >
          Github
        </a>
        <a
          class="text-well-read-400 underline"
          href="https://www.linkedin.com/in/joseph-lawrence-paul"
        >
          LinkedIn
        </a>
      </div>
    </div>
  </div>
</body>
`;
